plonky2x/src/frontend/vars/byte.rs

```diff
@@ -133,6 +134,40 @@ impl ByteVariable {
     pub fn from_be_bits(bits: [BoolVariable; 8]) -> ByteVariable {
         ByteVariable(bits)
     }
+
+    /// Creates a ByteVariable from a Target.
+    pub fn from_target<L: PlonkParameters<D>, const D: usize>(
+        builder: &mut CircuitBuilder<L, D>,
+        byte_target: Target,
+    ) -> Self {
+        builder.api.range_check(byte_target, 8);
+
+        let le_bits: [Target; 8] = builder
+            .api
+            .low_bits(byte_target, 8, 8)
+            .iter()
+            .map(|x| x.target)
+            .collect_vec()
+            .try_into()
+            .expect("Expected 8 bits.  Should never happen");
+        let mut bool_variables: [BoolVariable; 8] = le_bits.map(|x| x.into());
+
+        // Need to reverse it to big endian
+        bool_variables.reverse();
+        ByteVariable::from_be_bits(bool_variables)
+    }
+
+    /// Creates a Target from a ByteVariable.
+    pub fn to_target<L: PlonkParameters<D>, const D: usize>(
```

Comment:
nit: should we have these be `to_variable` and `from_variable` since we're trying to basically never operate with targets?

Reply:
I renamed `to_target` to `as_target` to be more similar to this function:  https://github.com/succinctlabs/succinctx/blob/main/plonky2x/src/frontend/vars/byte.rs#L109.

Also, I actually disagree about renaming `from_target` to `to_variable`.  It doesn't seem very Rust-like.  Ideally, I wanted to implement the From trait:

`impl From<Target> for BytesVariable`

but I needed a builder handler to do some of the bit manipulation, so the next best option seemed to be creating a `from_target` method.

Reply:
I mean that instead of returning a target, it should return a `Variable` (like a wrapped `Target`) since in our world, we always try to work with `Variable`, not `Target`

Reply:
done